mii/batching/ragged_batching.py

```diff
@@ -536,51 +535,62 @@ def make_request(self,
+        assert max_length > prompt_length, "prompt_length must be less than max_length"
+        max_new_tokens = kwargs.pop("max_new_tokens", max_length - prompt_length)
         stream = kwargs.pop("stream", False)
         ignore_eos = kwargs.pop("ignore_eos", False)
         # TODO: Add back this check
         # if self.policy.get_length(uid) + len(token_ids) >= max_length:
         #    raise ValueError(f"Session {uid} has reached max length {max_length}.")
 
-        postprocess_config = kwargs.pop("postprocess_config", {})
-        accepted_keys = ("logit_processor", "sampler", "stop_criterion")
-        for key in postprocess_config.keys():
-            if key not in accepted_keys:
-                raise ValueError(
-                    f"Unknown postprocess_config keyword {key}. Accepted keywords are {accepted_keys}"
-                )
-        logit_processor = _create_postprocessor(
-            postprocess_config.get("logit_processor",
-                                   DEFAULT_LOGITS_PROCESSOR),
-            LOGITS_PROCESSORS,
-        )
-        sampler = _create_postprocessor(
-            postprocess_config.get("sampler",
-                                   DEFAULT_SAMPLER),
-            SAMPLERS)
-        stop_criterion = _create_postprocessor(
-            postprocess_config.get("stop_criterion",
-                                   DEFAULT_STOP_CRITERION),
-            STOP_CRITERIA,
-            {"tokenizer": self.tokenizer},
-        )
+        post_processing = []
+
+        top_p = kwargs.pop("top_p", 0.9)
+        top_p_name = f"TopP_{top_p}"
+        if top_p_name not in self._post_processors:
+            self._post_processors[top_p_name] = TopPLogitProcessor(top_p=top_p)
+        post_processing.append(top_p_name)
+
+        top_k = kwargs.pop("top_k", None)
+        if top_k is not None:
+            top_k_name = f"TopK_{top_k}"
+            if top_k_name not in self._post_processors:
+                self._post_processors[top_k_name] = TopKLogitProcessor(top_k=top_k)
+            post_processing.append(top_k_name)
+
+        temp = kwargs.pop("temperature", None)
+        if temp is not None:
+            temp_name = f"Temp_{temp}"
+            if temp_name not in self._post_processors:
+                self._post_processors[temp_name] = TemperatureLogitProcessor(
+                    temperature=temp)
+            post_processing.append(temp_name)
+
+        sampler_name = "Sampler"
+        if sampler_name not in self._post_processors:
+            self._post_processors[sampler_name] = LogitsSampler()
+        post_processing.append(sampler_name)
+
+        stop_name = "Stop"
+        if stop_name not in self._post_processors:
+            self._post_processors[stop_name] = EosGenerationStopCriterion(
+                tokenizer=self.tokenizer)
+        post_processing.append(stop_name)
 
         assert kwargs == {}, f"Unknown keyword arguments {kwargs}"
 
         return [
             RaggedRequest(
```

Comment:
Just for sharing information, we now decompose prompts in the scheduler with the dynamic decomposition (SplitFuse). So it doesn't make sense to return this `RaggedRequest` as a list. We can simplify this in another refactoring PR.